$OpenBSD: patch-libptytty_src_fdpass_C,v 1.1 2013/01/07 08:08:39 dcoppa Exp $

Unbreak utmp support. Fix error:
"protocol error while reading pty/tty fds from helper process, aborting."

--- libptytty/src/fdpass.C.orig	Sat May 19 03:57:47 2012
+++ libptytty/src/fdpass.C	Wed Jan  2 17:08:40 2013
@@ -103,7 +103,7 @@ ptytty::recv_fd (int socket)
 
   if (recvmsg (socket, &msg, 0) >  0
       && data                   == 0
-      && msg.msg_controllen     >= CMSG_SPACE (sizeof (int)))
+      && msg.msg_controllen     >= CMSG_LEN (sizeof (int)))
     {
       cmsghdr *cmsg = CMSG_FIRSTHDR (&msg);
 
